plutus-core/testlib/PlutusCore/Generators/PIR/Common.hs

```diff
@@ -0,0 +1,98 @@
+{-# LANGUAGE BangPatterns          #-}
+{-# LANGUAGE ConstraintKinds       #-}
+{-# LANGUAGE DataKinds             #-}
+{-# LANGUAGE DeriveAnyClass        #-}
+{-# LANGUAGE DeriveFunctor         #-}
+{-# LANGUAGE DerivingVia           #-}
+{-# LANGUAGE FlexibleContexts      #-}
+{-# LANGUAGE FlexibleInstances     #-}
+{-# LANGUAGE GADTs                 #-}
+{-# LANGUAGE ImportQualifiedPost   #-}
+{-# LANGUAGE LambdaCase            #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE NumericUnderscores    #-}
+{-# LANGUAGE OverloadedStrings     #-}
+{-# LANGUAGE PartialTypeSignatures #-}
+{-# LANGUAGE PatternSynonyms       #-}
+{-# LANGUAGE RecordWildCards       #-}
+{-# LANGUAGE TemplateHaskell       #-}
+{-# LANGUAGE TupleSections         #-}
+{-# LANGUAGE TypeApplications      #-}
+{-# LANGUAGE TypeFamilies          #-}
+{-# LANGUAGE UndecidableInstances  #-}
+{-# LANGUAGE ViewPatterns          #-}
+{-# OPTIONS_GHC -Wno-name-shadowing #-}
+{-# OPTIONS_GHC -Wno-incomplete-uni-patterns #-}
+{-# OPTIONS_GHC -Wno-partial-type-signatures #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
+{-# OPTIONS_GHC -Wno-redundant-constraints #-}
+
+
+module PlutusCore.Generators.PIR.Common where
+
+
+import Data.Map (Map)
+import Data.Map qualified as Map
+import Data.String
+import Test.QuickCheck
+import Text.PrettyBy
+
+import PlutusCore.Default
+import PlutusCore.Name
+import PlutusIR
+
+-- CODE REVIEW: where to put the stuff below? Can we refactor to the point where we don't need them?
+-- Currently we need these for shrinking, getting rid of them would be nice.
+deriving stock instance Eq (Term TyName Name DefaultUni DefaultFun ())
+deriving stock instance Eq (Binding TyName Name DefaultUni DefaultFun ())
+deriving stock instance Eq (VarDecl TyName Name DefaultUni DefaultFun ())
+deriving stock instance Eq (TyVarDecl TyName ())
+deriving stock instance Eq (Datatype TyName Name DefaultUni DefaultFun ())
+
+-- Some convenience definitions that make the code slightly more readable.
```

Comment:
I have mixed feelings about this stuff. We don't use it anywhere else in the codebase, so in practice it makes this module *less* readable for people who generally work on the codebase.

Reply:
... Isn't that an argument for making the rest of the codebase more readable by adding cleaner patterns?

Reply:
> I have mixed feelings about this stuff. We don't use it anywhere else in the codebase, so in practice it makes this module less readable for people who generally work on the codebase.

I've removed all pattern synonyms. I'm gonna miss `(:->)`, though (others were not super useful, it was trivial to either refactor the call sites or just remove them completely).